src/bin/dual-iir.rs

```diff
@@ -173,24 +190,50 @@ const APP: () = {
             fence(Ordering::SeqCst);
 
             for channel in 0..adc_samples.len() {
-                adc_samples[channel]
-                    .iter()
-                    .zip(dac_samples[channel].iter_mut())
-                    .map(|(ai, di)| {
-                        let x = f32::from(*ai as i16);
-                        let y = settings.iir_ch[channel]
+                match settings.output_mode[channel] {
+                    OutputMode::IirFilter => {
+                        adc_samples[channel]
                             .iter()
-                            .zip(iir_state[channel].iter_mut())
-                            .fold(x, |yi, (ch, state)| {
-                                ch.update(state, yi, hold)
-                            });
-                        // Note(unsafe): The filter limits must ensure that the value is in range.
-                        // The truncation introduces 1/2 LSB distortion.
-                        let y: i16 = unsafe { y.to_int_unchecked() };
-                        // Convert to DAC code
-                        *di = DacCode::from(y).0;
-                    })
-                    .last();
+                            .zip(dac_samples[channel].iter_mut())
+                            .map(|(ai, di)| {
+                                let x = f32::from(*ai as i16);
+                                let y = settings.iir_ch[channel]
+                                    .iter()
+                                    .zip(iir_state[channel].iter_mut())
+                                    .fold(x, |yi, (ch, state)| {
+                                        ch.update(state, yi, hold)
+                                    });
+                                // Note(unsafe): The filter limits must ensure that the value is in range.
+                                // The truncation introduces 1/2 LSB distortion.
+                                let y: i16 = unsafe { y.to_int_unchecked() };
+                                // Convert to DAC code
+                                *di = DacCode::from(y).0;
+                            })
+                            .last();
+                    }
+                    OutputMode::SignalGenerator => {
+                        // Do not generate the samples twice, or we may mess up phasing of the
+                        // signal generator. Instead, copy the previously-generated signal.
+                        // TODO: Is there a nicer way we can handle this edge case?
```

Comment:
@jordens Is there any reason that we would need signal generation capabilities on both DAC outputs, or can we restrict the generation to a single channel? If we can restrict it, this logic gets simpler, as we just have a single `OutputChannel` setting for the signal generator that is either None, DAC0, or DAC1.